packages/@pollyjs/adapter-fetch/src/index.js

```diff
@@ -1,34 +1,46 @@
 import Adapter from '@pollyjs/adapter';
 import { Fetch as FetchUtils } from '@pollyjs/utils';
 
-const nativeFetch = global.fetch;
 const { defineProperty } = Object;
+const STUB_META = Symbol();
 
 export default class FetchAdapter extends Adapter {
   static get name() {
     return 'fetch';
   }
 
+  get defaultOptions() {
+    return {
+      context: global
+    };
+  }
+
   onConnect() {
-    this.assert('Fetch global not found.', nativeFetch);
+    const { context } = this.options;
+
+    this.assert('Fetch global not found.', context.fetch);
```

Comment:
`!!(context && context.fetch)` 

Assert only checks for false or undefined. Might wanna open that up to any falsey value.